### PR TITLE
Resource bouwstroompunten extra fields

### PR DESCRIPTION
**Pull request by cvriel**
The bouwstroompunten has some extra fields in the output. In addition, it opens the opportunity to update the dag so it uses the latest operator solutions. Making it more consistent with other dags.

### CODE REVIEW

File: src/dags/bouwstroompunten.py [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
 
+import dsnparse
 import requests
 from airflow import DAG
 from airflow.models import Variable
-from airflow.operators.bash import BashOperator
-from airflow.operators.python import PythonOperator
-from airflow.providers.postgres.operators.postgres import PostgresOperator
+from airflow.operators.bash_operator import BashOperator
+from airflow.operators.postgres_operator import PostgresOperator
```
Comment (gkoller) [inline src/dags/bouwstroompunten.py:10]: The original imports were good. This changes it to deprecated ones.
You should use:
```
from airflow.operators.python import PythonOperator
from airflow.providers.postgres.operators.postgres import PostgresOperator
```

File: src/dags/bouwstroompunten.py [hunk 2]
```diff
@@ -89,89 +91,98 @@ def get_data():
...
-    # 10. RUN bundled CHECKS
+    # 9. RUN bundled CHECKS
     multi_checks = PostgresMultiCheckOperator(task_id="multi_check", checks=total_checks)
 
-    # 11. Grant database permissions
+    # 10. Grant database permissions
     grant_db_permissions = PostgresPermissionsOperator(task_id="grants", dag_name=dag_id)
 
 
 (
```
Comment (gkoller) [inline src/dags/bouwstroompunten.py:181]: This should be indented one level more. The workflow specification (how the tasks are linked together) should be part of the `with` statement.

File: src/dags/bouwstroompunten.py [hunk 3]
```diff
@@ -38,34 +41,33 @@
...
 
+db_conn: object = DatabaseEngine()
 
-# data connection
-def get_data():
-    """getting the the access token and calling the data endpoint"""
 
-    # get token
-    token_url = f"{base_url}/{auth_endpoint}"
+def get_token() -> Any:
```
Comment (gkoller) [inline src/dags/bouwstroompunten.py:47]: By returning `Any` you are basically turning type annotations off. If possible you should be more specific.

You seem to be doing `return token_load["jwt"]`. What is that. Is it:
* a Base64Url encode string? If so, annotated the function return type with `-> str`
* a decoded JWT token? If so, annotate the function return type with `-> Dict[str, str] (assuming it's a dict, don't know much about JWT)

File: src/dags/bouwstroompunten.py [hunk 4]
```diff
@@ -89,89 +91,98 @@ def get_data():
     mkdir = BashOperator(task_id="mkdir", bash_command=f"mkdir -p {tmp_dir}")
```
Comment (gkoller) [inline src/dags/bouwstroompunten.py:91]: If you are using more modern operators might I suggest you you the `mk_dir` task here?

```python
from common.path import mk_dir
...
mkdir = mk_dir(tmp_dir, clean_if_exists=True)
```
Not sure if you want it cleaned though. If not, leave the argument out (default value is False), or set it explicitly to false.

The task `mk_dir` is our first usage of the Airflow 2.0 Flow based API. Much more convenient API than the Operator based one. See also the `anpr` DAG that already uses `mk_dir`

File: src/dags/bouwstroompunten.py (hunk 3 above)
Comment (gkoller) [reply src/dags/bouwstroompunten.py:47]: I suspect is should be `-> str` not that I have seen how it is used in:

```python
...
"Authorization": f"Bearer {get_token()}",
...
```

File: src/dags/bouwstroompunten.py [hunk 5]
```diff
@@ -89,89 +91,98 @@ def get_data():
...
+        mode="PostgreSQL",
+        db_conn=db_conn,
     )
 
-    # 6. Drop Exisiting TABLE
+    # 5. Drop Exisiting TABLE
     drop_tables = PostgresOperator(
         task_id="drop_existing_table",
         sql=[
             f"DROP TABLE IF EXISTS {dag_id}_{dag_id} CASCADE",
```
Comment (gkoller) [inline src/dags/bouwstroompunten.py:123]: I know this is existing code, but as you are touching the DAG anyway ...

We should never be using regular string formatting for creating SQL strings. Fortunately we are not processing user input here. Hence the potential for SQL injection is extremely low. But code gets copied around or mimicked hence it should be done properly.

The PostgresqlOperator has its own SQL string formatting capabilities. So it is best to use those:

```python
    drop_tables = PostgresOperator(
        task_id="drop_existing_table",
        sql=[
            "DROP TABLE IF EXISTS {{ params.table_id }} CASCADE",
        ],
        params={"table_id": f"{dag_id}_{dag_id}"} 
    )
```

**Review by gkoller — APPROVED** [review]
Couple of review to be addressed, but otherwise good to go :+1:

File: src/dags/bouwstroompunten.py (hunk 1 above)
Comment (cvriel) [reply src/dags/bouwstroompunten.py]: :+1:

File: src/dags/bouwstroompunten.py (hunk 2 above)
Comment (cvriel) [reply src/dags/bouwstroompunten.py]: :+1:

File: src/dags/bouwstroompunten.py (hunk 4 above)
Comment (cvriel) [reply src/dags/bouwstroompunten.py]: Nice addition, didn't know that!

File: src/dags/bouwstroompunten.py (hunk 5 above)
Comment (cvriel) [reply src/dags/bouwstroompunten.py]: Ok check!

**Pull request merged**